grid2op/tests/test_timeOutEnvironment.py

```diff
@@ -0,0 +1,149 @@
+        params = self.env1.parameters
+        params.NO_OVERFLOW_DISCONNECTION = True
+        self.env1.change_parameters(params)
+
+    def test_no_dn(self):
+        agentok = AgentOK(self.env1)
+        obs = self.env1.reset()
+        assert self.env1._nb_dn_last == 0
+        for i in range(10):
+            act = agentok.act(None, None, None)
+            obs, reward, done, info = self.env1.step(act)
+            assert info["nb_do_nothing"] == 0
+            assert info["nb_do_nothing_made"] == 0
+            assert self.env1._nb_dn_last == 0
+            
+    def test_one_dn(self):
+        agentko = AgentKO(self.env1)
+        obs = self.env1.reset()
+        assert self.env1._nb_dn_last == 0
+        for i in range(10):
+            act = agentko.act(None, None, None)
+            obs, reward, done, info = self.env1.step(act)
+            assert info["nb_do_nothing"] == 1
+            assert info["nb_do_nothing_made"] == 1
+            assert self.env1._nb_dn_last == 1
+            
+    def test_one_dn2(self):
+        agentko = AgentKO1(self.env1)
+        obs = self.env1.reset()
+        assert self.env1._nb_dn_last == 0
+        for i in range(10):
+            act = agentko.act(None, None, None)
+            obs, reward, done, info = self.env1.step(act)
+            assert info["nb_do_nothing"] == 1
+            assert info["nb_do_nothing_made"] == 1
+            assert self.env1._nb_dn_last == 1
+
+    def test_two_dn(self):
+        agentko2 = AgentKO2(self.env1)
+        obs = self.env1.reset()
+        assert self.env1._nb_dn_last == 0
+        for i in range(10):
+            act = agentko2.act(None, None, None)
+            obs, reward, done, info = self.env1.step(act)
+            assert info["nb_do_nothing"] == 2
+            assert info["nb_do_nothing_made"] == 2
+            assert self.env1._nb_dn_last == 2
+    
+    def test_diff_dn(self):
+        agentok = AgentOK(self.env1)
+        agentko = AgentKO(self.env1)
+        agentko2 = AgentKO2(self.env1)
+        obs = self.env1.reset()
+        assert self.env1._nb_dn_last == 0
+        for i, agent in enumerate([agentok, agentko, agentko2] * 2):
+            act = agent.act(None, None, None)
+            obs, reward, done, info = self.env1.step(act)
+            assert info["nb_do_nothing"] == i % 3
+            assert info["nb_do_nothing_made"] == i % 3
+            assert self.env1._nb_dn_last == i % 3
+        
+
+class TestTimedOutEnvironment50(TestTimedOutEnvironment100):
+    def get_timeout_ms(self):
+        return 50
+
+
+class TestTimedOutEnvironmentCpy(TestTimedOutEnvironment100):
+    def setUp(self) -> None:
+        super().setUp()
+        self.env0 = self.env1
+        self.env1 = self.env0.copy()
+        
+
+# TODO test runner
+# TODO test when used in gym (GymEnv)
+# TODO test that the "obs.simulate" and obs.get_forecast_env does not "do nothing"
+# TODO think about other tests
```

Comment:
These tests need to be implemented

Reply:
needs further explanation of what is asked, what is gym, the link between obs and action space... We will see this after my holidays, in May !  😎 😝